src/wallet/rpcdump.cpp

```diff
@@ -756,8 +756,13 @@ UniValue dumpwallet(const JSONRPCRequest& request)
 
     EnsureWalletIsUnlocked(pwallet);
 
-    fs::path filepath = request.params[0].get_str();
-    filepath = fs::absolute(filepath);
+    fs::path filepath = fs::absolute(request.params[0].get_str());
+    UniValue check = CanWriteFile(filepath.string());
+    int code = find_value(check, "code").get_int();
+
+    if (code != 0) {
+      throw check;
```

Comment:
Same here: `JSONRPCError` expected?

Reply:
Same

Reply:
But will now convert make the function return void and only throw, when irregular things happen (filename irregular, readonly mount etc.)